### PR TITLE
Feat/userdatamodel

### PR DESCRIPTION
**Pull request by thanh-nguyen-dang**

### CODE REVIEW

File: cloud_middleware/api/aws.py [hunk 1]
```diff
@@ -19,21 +19,25 @@
 import cloud_middleware.misc.constants as constants
 
 from auth import login_required
+from cloud_middleware.api.errors import InternalError
 from userdatamodel.user import User, ComputeAccess, ComputeProvider
 from flask import g, stream_with_context, Response, jsonify
 from flask import current_app as capp
 
 from sqlalchemy import and_
+from sqlalchemy.dialects.postgresql import JSONB
```
Comment (philloooo) [inline cloud_middleware/api/aws.py]: dead code?

File: cloud_middleware/api/aws.py [hunk 2]
```diff
@@ -71,74 +77,103 @@ def list_images(data):
...
                  ComputeAccess.user_id == user.id)).first()
-        if not compute_access:
-            compute_access = ComputeAccess(user_id=user.id, provider_id=provider.id)
-        compute_access.provider_vpc_info = {"security_group_ids": security_group_ids,
-                                            "subnet_id": subnet_id}
-        s.flush(compute_access)
+        if compute_access is None:
+            raise InternalError("No arn_role is set for this user at aws")
+        compute_access.additional_info["security_group_ids"] = security_group_ids
+        compute_access.additional_info["subnet_id"] = subnet_id
```
Comment (philloooo) [inline cloud_middleware/api/aws.py]: what's the security group ids and subnet id? why do we need to store this in compute_access?

File: cloud_middleware/api/aws.py [hunk 3]
```diff
@@ -71,74 +77,103 @@ def list_images(data):
...
         s.commit()
 
-@rest.post('/setArnRole')
+
+@rest.post('/set_arn_role')
```
Comment (philloooo) [inline cloud_middleware/api/aws.py]: this endpoint already exists in user-api, so no need to duplicate it in cloud_middleware

File: cloud_middleware/api/aws.py (hunk 2 above)
Comment (philloooo) [inline cloud_middleware/api/aws.py]: the code to get compute_access is duplicated several times, prolly just create a `get_compute_access(user, session)` function?

File: cloud_middleware/cloud_driver/aws_driver.py [hunk 4]
```diff
@@ -46,11 +46,12 @@ def login_with_arn_role(self, arn_role, service='sts'):
         return ec2, credentials
 
     @uuid_quickfail
-    def delete_instance(self, instance_id):
+    def delete_instances(self, arn_role, instance_ids):
         """ Delete cluster that has instance with instance_id as a member """
```
Comment (philloooo) [inline cloud_middleware/cloud_driver/aws_driver.py]: what do you mean by `Delete cluster`?

File: cloud_middleware/api/aws.py [hunk 5]
```diff
@@ -71,74 +77,103 @@ def list_images(data):
...
                                                    max_count=int(server_data["count"]),
                                                    min_count=int(server_data["count"]),
                                                    **server_data)
     return list_instances_to_json(instances)
 
+
 def update_vpc(security_group_ids, subnet_id):
     with capp.db.session as s:
         user = s.query(User).filter(User.username == g.user.username).first()
         provider = s.query(ComputeProvider).filter(ComputeProvider.name == 'aws').first()
```
Comment (philloooo) [inline cloud_middleware/api/aws.py]: `s/aws/AWS/g` to match with userapi?

File: cloud_middleware/api/aws.py (hunk 1 above)
Comment (thanh-nguyen-dang) [inline cloud_middleware/api/aws.py]: yes

File: cloud_middleware/api/aws.py (hunk 2 above)
Comment (thanh-nguyen-dang) [inline cloud_middleware/api/aws.py]: sure, I will add it. We need have security_group_ids the spin up vms with a vpc

File: cloud_middleware/cloud_driver/aws_driver.py (hunk 4 above)
Comment (thanh-nguyen-dang) [inline cloud_middleware/cloud_driver/aws_driver.py]: I mean delete instances (vms)

File: cloud_middleware/api/aws.py (hunk 5 above)
Comment (thanh-nguyen-dang) [inline cloud_middleware/api/aws.py]: what do you mean? what is s/aws/AWS/g?

**Pull request merged**